modules/mono/editor/GodotTools/GodotTools.ProjectEditor/ProjectGenerator.cs

```diff
@@ -25,6 +25,19 @@ public static ProjectRootElement GenGameProject(string name)
             mainGroup.AddProperty("TargetFramework", "net6.0");
             mainGroup.AddProperty("EnableDynamicLoading", "true");
 
+            if (!string.IsNullOrEmpty(projectDir))
+            {
+                mainGroup.AddProperty("GodotProjectDir", Path.Combine("$(SolutionDir)", projectDir));
+
+                string basePath = Path.Combine("$(GodotProjectDir)", ".godot", "mono", "temp", "bin");
+
+                mainGroup.AddProperty("BaseOutputPath", basePath);
+                mainGroup.AddProperty("OutputPath", Path.Combine(basePath, "$(Configuration)"));
+
+                var itemGroup = root.AddItemGroup();
+                itemGroup.AddItem("Compile", Path.Combine(projectDir, "*.cs"));
+            }
+
```

Comment:
No, this should not be generated as part of the C# project file. This problem should be fixed in the Sdk:

https://github.com/godotengine/godot/blob/cd491c6e47bb41ccf596189676c0c5d4a72226cc/modules/mono/editor/Godot.NET.Sdk/Godot.NET.Sdk/Sdk/Sdk.props#L15-L16

Can you check if changing those two lines to the following works?

```csproj
<GodotProjectDir Condition=" '$(GodotProjectDir)' == '' ">$(MSBuildProjectDirectory)</GodotProjectDir>
```

Reply:
Seems not. `$(MSBuildProjectDirectory)` indicates the path of `csproj` files. If users don't put the `csproj` file in the default directory, the engine still could not find the `dll` file.

Reply:
Do you have any project with the `csproj` in a different location? Do you know anyone else doing that? I think it was a mistake from my part to allow that, so I'm considering reverting it. Allowing to change the solution location is fine, but the `csproj` should be in the same folder as `project.godot`. The setting should only allow changing the project name, not the project directory.

Reply:
> Do you have any project with the `csproj` in a different location? Do you know anyone else doing that?

Of course no. 

I agree with putting `csproj` where `project.godot` is. That might be the best solution so far. I will revoke this part of modification sooner.

Reply:
I also removed `csproj` settings. Maybe we need to change the title of this PR.

Reply:
Looks good!

Reply:
Hi, I'm just one person and I'm prepared to be misunderstanding this decision, or wrong about my approach, or an uncommon case that's not worth the effort. However, I am actually putting (or trying to put) the csproj in a different location than the godot project, because I'm creating multiple csprojs in a solution and I'm used to organizing multiple projects in subdirectories of a solution. (E.g., [this example](https://gist.github.com/davidfowl/ed7564297c61fe9ab814) of a solution layout.)

I appreciate the fix, but I guess I don't understand what the purpose of moving the sln would be, if you can't also move the csproj. You end up with the sln, which is the organizational root, in a subdirectory, but the csproj, which is below the sln in the solution hierarchy, above the sln in the directory structure. To me this seems like it undermines good solution organization, especially if you have other csprojs that will live in sub-subdirectories under the sln.

Thanks for the fix and all the other work developing Godot!

Reply:
The example solution layout that you shared should still work, no need to change the csproj location.

Maybe take a look at the layout in issue https://github.com/godotengine/godot/issues/69225 for an example that should work for multi-project scenarios.

I think your misunderstanding comes from using the Godot project directory as the root, when it should be just one more C# project in the solution.